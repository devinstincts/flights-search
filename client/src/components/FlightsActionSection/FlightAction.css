.flightActionsContainer {
    height: 50px;
    padding: 15px;
    background: #fff;
    color: #00b2d6;
    font-size: 18px;
    display: flex;
}
.flightSortFilterActionContainer {
    display: flex;
    align-items: center;
    flex: 1;
}
.flight-filter-sort-action {
    text-align: left;
    flex: 1;
    cursor: pointer;
}
.flightPriceAlterContainer {
    flex: 1;
    text-align: right;
    justify-self: flex-end;
    display: inline-flex;
    align-items: center;
    justify-content: flex-end;
    cursor: pointer;
}
.flight-action-image {
    height: 18px;
    width: 18px;
    margin-right: 8px;
    display: inline-block;
}
