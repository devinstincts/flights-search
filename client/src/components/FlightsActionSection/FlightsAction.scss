@import '~bpk-mixins';

.FlightsAction {
    display: flex;
    justify-content: space-between;
    color: #00b2d6;
    margin: 0 0 10px 0;

    &__filter-section {
        span {
            margin: 0 35px 0 5px;
            font-weight: 600;
        }
    }

    &__alert-image {
      height: $bpk-spacing-base;
      vertical-align: top;
      height: 18px;
      width: 18px;
      margin-right: 5px;
    }
}