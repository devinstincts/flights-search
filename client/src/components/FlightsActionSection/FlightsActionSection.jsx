import React from 'react';
import alertIcon from './notification.svg';
import './FlightAction.css';

const FlightsActionSection = () => (
  <div className="flightActionsContainer">
    <div className="flightSortFilterActionContainer">
      <span className="flight-filter-sort-action">Filter</span>
      <span className="flight-filter-sort-action">Sort</span>
    </div>
    <div className="flightPriceAlterContainer">
      <img className="flight-action-image" alt="Alerts" src={alertIcon} />
      <span>Price alerts</span>
    </div>
  </div>
);

export default FlightsActionSection;
