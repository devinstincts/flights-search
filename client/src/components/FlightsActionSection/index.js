import FlightsActionSection from './FlightsActionSection';

export default FlightsActionSection;
