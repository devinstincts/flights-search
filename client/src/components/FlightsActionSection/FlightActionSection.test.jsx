import React from 'react';
import ReactDOM from 'react-dom';
import ShallowRenderer from 'react-test-renderer/shallow';
import FlightsActionSection from './index';

describe('FlightsActionSection', () => {
  it('should render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<FlightsActionSection />, div);
  });

  it('should render correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(<FlightsActionSection />);
    expect(renderer.getRenderOutput()).toMatchSnapshot();
  });
});
