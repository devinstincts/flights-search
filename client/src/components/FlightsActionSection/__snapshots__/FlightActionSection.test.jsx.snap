// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlightsActionSection should render correctly 1`] = `
<div
  className="flightActionsContainer"
>
  <div
    className="flightSortFilterActionContainer"
  >
    <span
      className="flight-filter-sort-action"
    >
      Filter
    </span>
    <span
      className="flight-filter-sort-action"
    >
      Sort
    </span>
  </div>
  <div
    className="flightPriceAlterContainer"
  >
    <img
      alt="Alerts"
      className="flight-action-image"
      src="notification.svg"
    />
    <span>
      Price alerts
    </span>
  </div>
</div>
`;
